File: postgres_39/struct/RmSeisTrace.go
package dto

import (
	"time"
)

type Rm_seis_trace struct {
	Info_item_subtype         string     `json:"info_item_subtype"`
	Information_item_id       string     `json:"information_item_id"`
	Active_ind                *string    `json:"active_ind"`
	Datum_vel_correction      *string    `json:"datum_vel_correction"`
	Datum_vel_correction_ouom *string    `json:"datum_vel_correction_ouom"`
	Effective_date            *time.Time `json:"effective_date"`
	Expiry_date               *time.Time `json:"expiry_date"`
	First_sample_time         *float64   `json:"first_sample_time"`
	First_sample_timezone     *string    `json:"first_sample_timezone"`
	Horizontal_scale          *string    `json:"horizontal_scale"`
	Horizontal_scale_ouom     *string    `json:"horizontal_scale_ouom"`
	Phase                     *string    `json:"phase"`
	Polarity                  *string    `json:"polarity"`
	Ppdm_guid                 string     `json:"ppdm_guid"`
	Record_length             *float64   `json:"record_length"`
	Record_length_ouom        *string    `json:"record_length_ouom"`
	Reference_datum           *float64   `json:"reference_datum"`
	Reference_datum_ouom      *string    `json:"reference_datum_ouom"`
	Remark                    *string    `json:"remark"`
	Replacement_vel           *float64   `json:"replacement_vel"`
	Replacement_vel_ouom      *string    `json:"replacement_vel_ouom"`
	Reported_first_ref_num    *string    `json:"reported_first_ref_num"`
	Reported_last_ref_num     *string    `json:"reported_last_ref_num"`
	Reported_ref_num_type     *string    `json:"reported_ref_num_type"`
	Sample_rate               *float64   `json:"sample_rate"`
	Sample_rate_ouom          *string    `json:"sample_rate_ouom"`
	Sample_type               *string    `json:"sample_type"`
	Source                    *string    `json:"source"`
	Static_correction         *float64   `json:"static_correction"`
	Static_correction_ouom    *string    `json:"static_correction_ouom"`
	Variable_area_ind         *string    `json:"variable_area_ind"`
	Vertical_scale            *string    `json:"vertical_scale"`
	Vertical_scale_ouom       *string    `json:"vertical_scale_ouom"`
	Row_changed_by            *string    `json:"row_changed_by"`
	Row_changed_date          *time.Time `json:"row_changed_date"`
	Row_created_by            *string    `json:"row_created_by"`
	Row_created_date          *time.Time `json:"row_created_date"`
	Row_effective_date        *time.Time `json:"row_effective_date"`
	Row_expiry_date           *time.Time `json:"row_expiry_date"`
	Row_quality               *string    `json:"row_quality"`
}

File: create_tables/rm_seis_trace.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table rm_seis_trace';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute(
    """create table rm_seis_trace(
        INFO_ITEM_SUBTYPE VARCHAR(30) NOT NULL,
        INFORMATION_ITEM_ID VARCHAR(40) NOT NULL,
        ACTIVE_IND VARCHAR(1),
        DATUM_VEL_CORRECTION VARCHAR(240),
        DATUM_VEL_CORRECTION_OUOM VARCHAR(40),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        FIRST_SAMPLE_TIME DECIMAL(10,5),
        FIRST_SAMPLE_TIMEZONE VARCHAR(40),
        HORIZONTAL_SCALE VARCHAR(40),
        HORIZONTAL_SCALE_OUOM VARCHAR(40),
        PHASE VARCHAR(40),
        POLARITY VARCHAR(40),
        PPDM_GUID VARCHAR(38),
        RECORD_LENGTH DECIMAL(10,5),
        RECORD_LENGTH_OUOM VARCHAR(40),
        REFERENCE_DATUM DECIMAL(10,5),
        REFERENCE_DATUM_OUOM VARCHAR(40),
        REMARK VARCHAR(2000),
        REPLACEMENT_VEL DECIMAL(10,5),
        REPLACEMENT_VEL_OUOM VARCHAR(40),
        REPORTED_FIRST_REF_NUM VARCHAR(30),
        REPORTED_LAST_REF_NUM VARCHAR(30),
        REPORTED_REF_NUM_TYPE VARCHAR(40),
        SAMPLE_RATE DECIMAL(8,4),
        SAMPLE_RATE_OUOM VARCHAR(40),
        SAMPLE_TYPE VARCHAR(40),
        SOURCE VARCHAR(40),
        STATIC_CORRECTION DECIMAL(10,5),
        STATIC_CORRECTION_OUOM VARCHAR(40),
        VARIABLE_AREA_IND VARCHAR(1),
        VERTICAL_SCALE VARCHAR(40),
        VERTICAL_SCALE_OUOM VARCHAR(40),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")